### PR TITLE
General Griefsky

### PR DESCRIPTION
**Pull request by Nicaragua1**
Meet General Griefsky!
Inspired from TG. (general grievous)
This bot is powerful. If you manage to get 4 eswords you can construct it and make the station the safest place in the galaxy

This bot will mercilessly track, kill, maim and subsequently disembowel his victims.

newgriefskyyy
(sprites made by: Danaleja2005) https://github.com/Danaleja2005

craft

griefskymoves
_(It has a toy version! left one) The sprite from the right is  new !. Apologies for the bad FPS of the gif , i suggest downloading the dmi  to see its actual spinning_

iconsgriefsky

**NOTE:** Its important to say that despide of its name , Griefsky is an advanced security bot. This means without emagging it, it works like a regular secbot.  That being said, you need 4 e-swords plus an emag if you are planning on using it for chaos or distraction purposes.

Emagged griefskys have a red light to warn people before its too late

definitive

**GRIEFSKY STATS:**
-Health 100 (75+ than beepsky)
-50% melee block chance
-90% blocking ranged attacks
-50% chance of stunning
-Approachs targets 0,1 faster than beepsky 

**Things to be done:**
- Balancing stats! (depending on community feedback plus admins knowledge)

:cl:
add: Adds General Griefsky bot and a toy version!
soundadd: Spin-sabre sound
imageadd: Adds Griefsky sprites
/:cl:

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
Hello there.

**Comment by Diegoflores31** [comment]
The True Greytide terror

**Comment by Nicaragua1** [comment]
I think griefsky aprentice should be construction only and removed from recipes. Like colored EDs .

**Comment by DesolateG** [comment]
Can you make a toy esword one?

**Comment by Nicaragua1** [comment]
> Can you make a toy esword one?

yes

**Comment by DesolateG** [comment]
That's going to be adorable

**Comment by Nicaragua1** [comment]
done

**Comment by Nicaragua1** [comment]
> Can you make a toy esword one?

done

**Comment by Spartan6** [comment]
Does it have custom voice lines?

**Comment by Nicaragua1** [comment]
@Spartan6 yes it does:smiley:

**Comment by FlattestGuitar** [comment]
1. Why isn't this entire thing just inheriting from beepsky?
2. Hate to say it, but it definitely looks like you're trying to make it seems like it's all your own work, when it's just a port of a /tg/ PR made by KMC. <https://github.com/tgstation/tgstation/pull/39382>

**Comment by Pixelzilla** [comment]
I dont really see why we need what basically boils down to a validhunter's dream bot? It doesn't seem to fit in with the sort of playstyle paradise is trying to have. I cant really think of many situations where this bot could be constructed and used for anything other than its namesake, griefing.

Its likely uses are going to be from security breaking SOP to have one built in the first place to hunt down anyone wanted and likely end up murdering innocents along the way (at which point...who's responsible?). Even if traitors make it the fine line traitors are required to walk to only do their objectives make this bot likely to result in everyone except hijack-related traitors being bwoinked.

Another concern....this bot is not able to be pAI controlled without admin intervention like the other secbots, right?

**Comment by Nicaragua1** [comment]
@FlattestGuitar 1. It has too many differences with a secbot to be a sub type of it

**Comment by Dinarzad** [comment]
> I dont really see why we need what basically boils down to a validhunter's dream bot? It doesn't seem to fit in with the sort of playstyle paradise is trying to have. I cant really think of many situations where this bot could be constructed and used for anything other than its namesake, griefing.
> 
> Its likely uses are going to be from security breaking SOP to have one built in the first place to hunt down anyone wanted and likely end up murdering innocents along the way (at which point...who's responsible?). Even if traitors make it the fine line traitors are required to walk to only do their objectives make this bot likely to result in everyone except hijack-related traitors being bwoinked.
> 
> Another concern....this bot is not able to be pAI controlled without admin intervention like the other secbots, right?

Thing is, getting 4 e-swords is gonna be a major stopping point. Not every traitors buys them, especially the more experienced a player is an they start looking for more challenges.  Crew making this is gonna be pretty rare.
Traitors making it is also gonna be infrequent unless they make the dagger version, in which case they just spent a large chunk of telecrystals on a meme that Security comes pre-equipped to handle (Ion rifle).  It catches a few people off guard, Security handles it, those people get cloned, the traitor has pushed station to red alert, and is hunted down actively.  I don't see a lot of people doing it aside for the meme value.

Additionally this still requires the helmet from security and robot arms to make, which means it's almost entirely gonan be made by traitor roboticists that are completely giving themselves away. Anyone NOT a roboticist is probably not gonna get a sec helmet without jumping and killing an officer as it is.  I think we are drastically over stating how much of a problem this bot is going to be, it's something that, after the initial fad of it fades, will be seen once in a while at best.

An frankly, more people could do with a few "school of hard knocks" lessons about dying in SS13. It happens.

**Comment by Shadeykins** [comment]
> 
> 
> > I dont really see why we need what basically boils down to a validhunter's dream bot? It doesn't seem to fit in with the sort of playstyle paradise is trying to have. I cant really think of many situations where this bot could be constructed and used for anything other than its namesake, griefing.
> > Its likely uses are going to be from security breaking SOP to have one built in the first place to hunt down anyone wanted and likely end up murdering innocents along the way (at which point...who's responsible?). Even if traitors make it the fine line traitors are required to walk to only do their objectives make this bot likely to result in everyone except hijack-related traitors being bwoinked.
> > Another concern....this bot is not able to be pAI controlled without admin intervention like the other secbots, right?
> 
> Thing is, getting 4 e-swords is gonna be a major stopping point. Not every traitors buys them, especially the more experienced a player is an they start looking for more challenges and this would require you to catch 4 seperate traitors with each having an E-sword. Security doesn't make that quota very often as it is. Crew making this is gonna be pretty rare.
> 
> Traitors making it is also gonna be infrequent unless they make the dagger version or work together on, in which case they just spent a large chunk of telecrystals on a meme that Security comes pre-equipped to handle (Ion rifle). It catches a few people off guard, Security handles it, those people get cloned, the traitor has pushed station to red alert, and is hunted down actively. I don't see a lot of people doing it aside for the meme value.
> 
> Additionally this still requires the helmet from security and robot arms to make, which means it's almost entirely gonan be made by traitor roboticists that are completely giving themselves away. Anyone NOT a roboticist is probably not gonna get a sec helmet without jumping and killing an officer as it is. I think we are drastically over stating how much of a problem this bot is going to be, it's something that, after the initial fad of it fades, will be seen once in a while at best.
> 
> An frankly, more people could do with a few "school of hard knocks" lessons about dying in SS13. It happens.

To play devil's advocate, all it takes is two traitors getting together and they can make this rather easily.

**Comment by KasparoVy** [comment]
A nerf to enjoyment of the game. Interesting idea and cool sprites, though...

**Comment by Dinarzad** [comment]
> To play devil's advocate, all it takes is two traitors getting together and they can make this rather easily.

Fair enough. 
But even then, that takes a partnership of 2 traitors, wasting TC to make something not all that effective in the grand scheme, at least, not unless they *also* steal the ion rifle (Assuming they rush this out ASAP anyway, if it's later in the round EMPs are easy enough to make from a myriad of sources). It would take a little bit of prep work to get the station into a position where this seems like it might be really lethal, at least any more so then what you could have gotten for just buying a holopara-buddy to help you murderate. It seems like, at best, a really good distraction and at worst a giggle between two trators.

And failing ion rifles being on hand, there's always good ol' floor tile ninja stars. It's a little hard to tell from the gif, but it doesn't SEEM like it moves much faster then ol' beepers himself, so kiting it should be easy enough. Unsure if block chance would apply to greytide ninja stars, but you could just keep doing it until it sticks. It really seems like it's mostly meme value for laughs then any kind of serious threat.

**Comment by taukausanake** [comment]
I'd get rid of the edagger variant to be honest. Doesn't sound that good. You are not going to be able to hit this guy easily unless it has collision like what Dalto mentioned.

Otherwise [_yeeeeeeeesssssssssss_](https://github.com/ParadiseSS13/Paradise/pull/10672#issuecomment-455793341)

File: code/modules/mob/living/simple_animal/bot/griefsky.dm [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+	block_chance_ranged = 50
+	spin_icon = "griefskyj-c"
+	window_name = "Automatic Security Unit v7.0"
+	health = 100
+	maxHealth = 100
+	weapon = /obj/item/pen/edagger
+	
+
+/mob/living/simple_animal/bot/griefsky/griefsky/toy  //A toy version of general griefsky!
+	name = "Genewul Bweepskee"
```
Comment (taukausanake) [inline code/modules/mob/living/simple_animal/bot/griefsky.dm]: I would think the toy version name would be similar to the new griefy one but I don't know. I would have done Gweafskee I think

**Comment by Nicaragua1** [comment]
This Pull Request will be closed to remove e daggers version and probably remake all the code to be a subtype of secbot.

Until then...

File: code/modules/mob/living/simple_animal/bot/griefsky.dm [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+	var/threat = C.assess_threat(src)
+	if(ishuman(C))
+		C.apply_damage(dmg, BRUTE)
+		if(prob(stun_chance)) 
+			C.Weaken(5)
+	add_attack_logs(src, C, "sliced")
+	if(declare_arrests)
+		var/area/location = get_area(src)
+		if(!spam_flag)
+			speak("Back away! i will deal with this level [threat] swine <b>[C]</b> in [location] myself!.", radio_channel)
```
[the remark this answers is not in the document]
Comment (Nicaragua1) [reply code/modules/mob/living/simple_animal/bot/griefsky.dm]: thanks

**Comment by Nicaragua1** [comment]
E-dagger version was removed entirely. Now is inheriting from secbots. This saves 200 lines of code.

Griefsky stats should be looked at and find a reasonable level of balance

File: code/modules/mob/living/simple_animal/bot/griefsky.dm [hunk 3]
```diff
@@ -0,0 +1,223 @@
+/mob/living/simple_animal/bot/secbot/griefsky //This bot is powerful. If you managed to get 4 eswords somehow, you deserve this horror. Emag him for best results.
+	name = "General Beepsky"
```
Comment (taukausanake) [inline code/modules/mob/living/simple_animal/bot/griefsky.dm]: Double check that name please? Everything else is named Griefsky

**Comment by Nicaragua1** [comment]
The PR is ready to be merged if accepted.

 Of course balancing is still pending

**Comment by davidchan13** [comment]
Amusing as this is, seems much more something nuke ops or hijack traitors would make, not regular crew.

**Pull request merged**